aptos-move/aptos-debugger/src/lib.rs

```diff
@@ -72,33 +72,36 @@ impl AptosDebugger {
             .check_signature()
             .map_err(|err| format_err!("Unexpected VM Error: {:?}", err))?;
 
-        let (status, output, gas_profiler) =
-            AptosVM::execute_user_transaction_with_custom_gas_meter(
-                &state_view,
-                &txn,
-                &log_context,
-                |gas_feature_version, gas_params, storage_gas_params, balance| {
-                    let gas_meter =
-                        MemoryTrackedGasMeter::new(StandardGasMeter::new(StandardGasAlgebra::new(
-                            gas_feature_version,
-                            gas_params,
-                            storage_gas_params,
-                            balance,
-                        )));
-                    let gas_profiler = match txn.payload() {
-                        TransactionPayload::Script(_) => GasProfiler::new_script(gas_meter),
-                        TransactionPayload::EntryFunction(entry_func) => GasProfiler::new_function(
-                            gas_meter,
-                            entry_func.module().clone(),
-                            entry_func.function().to_owned(),
-                            entry_func.ty_args().to_vec(),
-                        ),
-                        TransactionPayload::ModuleBundle(..) => unreachable!("not supported"),
-                        TransactionPayload::Multisig(..) => unimplemented!("not supported yet"),
-                    };
-                    Ok(gas_profiler)
-                },
-            )?;
+        // TODO(Gas): revisit this.
+        let vm = AptosVM::new_from_state_view(&state_view);
+        let resolver = state_view.as_move_resolver();
+
+        let (status, output, gas_profiler) = vm.execute_user_transaction_with_custom_gas_meter(
+            &resolver,
```

Comment:
why do we need to pass resolver if we pass the vm as self now?

Reply:
But we should pass either that, or state view or executor view? Passing a resolver seems logical because it is mostly used anyway

Reply:
I think the question was whether you can just get resolver from the self (VM) inside execute_user_transaction_with_custom_gas_meter?

Reply:
VM does not contain a resolver. You can get one but then you need to pass at least `&state_view`